Server/notebook2html

gramster opened the pull request:
Makes the 'Save as HTML' have workable charts by inlining dependencies.

nikhilk commented on containers/datalab/Dockerfile.in:
```diff
@@ -106,8 +106,11 @@ ADD build/lib/GCPData-0.1.0.tar.gz /datalab/lib/
+ADD nbconvert /datalab/nbconvert
 ADD content/ /datalab
 ADD config/ipython.py /root/.ipython/profile_default/ipython_config.py
+RUN mkdir -p /root/.jupyter
```
Is this line needed - there is a RUN jupyter notebook --generate-config step earlier.

gramster replied:
Removed

nikhilk commented on containers/datalab/Dockerfile.in:
```diff
@@ -115,6 +118,16 @@ RUN cd /datalab/lib/GCPData-0.1.0 && python setup.py install && \
     cd /datalab/web && /tools/node/bin/npm install && \
     cd /
 
+# Link stuff needed for nbconvert to a location where Jinja will find it.
```
Turn this into a script and run in the same docker step as the previous RUN statement to minimize # of layers in our container.

gramster replied:
I could just fold this in with the previous RUN using && rather than adding a script. Thoughts?

nikhilk replied:
Sure, that works as well for now ... I feel its a bit short term (eg. if we add a file, we have to change it here), but we can defer it. I'd hope a script can do something like foreach file in dir1, create a link.

nikhilk commented on containers/datalab/config/ipython.py:
```diff
@@ -28,3 +28,7 @@
+
+# Customize the Save as HTML handling
+c.TemplateExporter.template_path = ['/datalab/nbconvert']
+c.TemplateExporter.template_file = 'as_html'
```
Somewhat of a nit -- but as_html as a file name seems odd. Just 'html'? But maybe there is more to it, in which case its fine.

nikhilk replied:
Also to confirm, is there a reason ipython.config needs this? I would have expected the kernel to know nothing about notebooks... maybe just config/nbconvert.py suffices?

gramster replied:
I can check again but IIRC this is actually the important one; the config/nbconvert.py only affects running nbconvert from the CLI if you shell into the container (which I did for testing).

nikhilk commented on sources/web/datalab/static/extensions/charting.js:
```diff
@@ -201,6 +201,12 @@ define(function () {
+    if (model.chartStyle == 'paged_table' && document._in_nbconverted) {
+       model.chartStyle = 'table';
+       var p = document.createElement("div");
+       p.innerHTML = '<br>(Truncated...)';
```
Suggest "Truncated" ->  "Truncated to first page of results"

gramster replied:
Done

nikhilk commented:
OK, comments added.

gramster commented:
This is no longer working and I'm not sure how I can get --debug output from Jupyter to show in the logs. If I copy the jupyter_notebook_config.py to jupyter_nbconvert_config.py and run jupyter nbconvert on the command line it is working correctly, but for some reason when running via the proxy server it is not picking up the config file anymore (and jupyter_notebook_config.py is the correct name; that much I know).

gramster commented:
Pushed an update addressing the comments - and it's working again today. I must have somehow got my build into a bad state yesterday; maybe related to the container build changes? Not sure but in between checking out and building master then checking out and building this branch again things started working.
